src/rate_limiter/src/lib.rs

```diff
@@ -180,24 +191,24 @@ impl TokenBucket {
         }
 
         if tokens > self.budget {
-            // TODO (Issue #259) remove this block when issue is resolved
+            // This operation requests a bandwidth higher than the bucket size
             if tokens > self.size {
                 error!(
-                    "Trying to consume more tokens {} than the total capacity {}",
+                    "Consumed {} tokens from bucket of size {}",
                     tokens, self.size
                 );
-                // best effort rate-limiting, this is a dirty workaround for Issue #259
-                if self.budget == self.size {
-                    self.budget = 0;
-                    return true;
-                }
+                // Empty the bucket and report an overconsumption of
+                // (remaining tokens / size) times larger than the bucket size
+                tokens -= self.budget;
+                self.budget = 0;
+                return BucketReduction::OverConsumption(tokens as f64 / self.size as f64);
```

Comment:
Can self.size be 0 ?

Reply:
No, as you cannot construct a `TokenBucket` with size 0.